debezium-connector-sqlserver/src/test/java/io/debezium/connector/sqlserver/SnapshotWithSelectOverridesIT.java

```diff
@@ -139,16 +139,17 @@ public void takeSnapshotWithOverrides() throws Exception {
     @Test
     @FixFor("DBZ-3429")
     public void takeSnapshotWithOverridesWithAdditionalWhitespace() throws Exception {
+        final String databaseName = TestHelper.TEST_DATABASE;
         final Configuration config = TestHelper.defaultConfig()
                 .with(
                         RelationalDatabaseConnectorConfig.SNAPSHOT_SELECT_STATEMENT_OVERRIDES_BY_TABLE,
                         "  dbo.table1 , dbo.table3  ")
                 .with(
                         RelationalDatabaseConnectorConfig.SNAPSHOT_SELECT_STATEMENT_OVERRIDES_BY_TABLE + ".dbo.table1",
-                        "SELECT * FROM [dbo].[table1] where soft_deleted = 0 order by id desc")
+                        "SELECT * FROM " + databaseName + ".[dbo].[table1] where soft_deleted = 0 order by id desc")
```

Comment:
This is a breaking change. Being able to override snapshot `SELECT` statements is part of the public API but the existing statements won't work if they don't contain fully qualified table names because the connection no longer uses any database.

We need to pause here and introduce the configuration for opting into the breaking changes.